### PR TITLE
VBID-8201 Add Email Validator

### PR DESCRIPTION
**Pull request by psimmons-vistar**
Created new Field that will validate that the provided value is
an email.
Also did a little cleanup in the validators code.

### CODE REVIEW

File: rest/validators.py [hunk 1]
```diff
@@ -27,20 +27,20 @@ def test_len(value):
...
     if value and not re.search(expr, value):
       return [msg]
 
   return contains_regex
 
+url = regex(r'^http(s)?://([^/:]+\.[a-z]{2,10}|' + \
+              r'([0-9]{1,3}\.){3}[0-9]{1,3})(:[0-9]+)?(\/.*)?$',
+            'must be a valid HTTP URL')
+
+email = regex(r'(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)',
```
Comment (nweinthal) [inline rest/validators.py]: Not sure if we're concerned about it, but I don't think this'd match unicode characters like ñ

Comment (robolange) [inline rest/validators.py]: Yeah, this is going to miss some addresses. For example, apostrophes are technically allowed in username part, and I have actually seen some before. I guess it depends on how accurate this needs to be, but the above will catch probably 99% of emails.

**Comment by markchadwick** [comment]
Aaaah, georest. Looks good to me

**Comment by gsf** [comment]
I agree with https://stackoverflow.com/a/8022584 -- make the regex even dumber.

**Comment by psimmons-vistar** [comment]
Reason I like what I went with is that it is just accurate enough to cover every account that we have in the system (account emails are custom @vistarmedia.com emails that Platform chooses).

I don't want to go with anything more complete for email validation since having it be more restricted will catch more potential user input errors in the Geo UI, which is the purpose of this for geo-rest.

If this library was shared with more active projects of ours I'd feel differently

**Pull request merged**